tests/e2e/storage_policy_utils.go

```diff
@@ -0,0 +1,153 @@
+				PropertyList: []pbm.Property{
+					{
+						ID:       "VolumeAllocationType",
+						Value:    "Fully initialized",
+						DataType: "string",
+					},
+				},
+			},
+		},
+	}
+	for k, v := range categoryTagMap {
+
+		pbmCreateSpec.CapabilityList = append(pbmCreateSpec.CapabilityList, pbm.Capability{
+			ID:        k,
+			Namespace: "http://www.vmware.com/storage/tag",
+			PropertyList: []pbm.Property{
+				{
+					ID:       "com.vmware.storage.tag." + k + ".property",
+					Value:    v,
+					DataType: "set",
+				},
+			},
+		})
+	}
+	createSpecVMFS, err := pbm.CreateCapabilityProfileSpec(pbmCreateSpec)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	profileID, err := pbmClient.CreateProfile(ctx, *createSpecVMFS)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	framework.Logf("VMFS profile with id: %v and name: '%v' created", profileID.UniqueId, profileName)
+
+	return profileID, profileName
+}
+
+//deleteStoragePolicy deletes the given storage policy
+func deleteStoragePolicy(ctx context.Context, pbmClient *pbm.Client, profileID *types.PbmProfileId) {
+	_, err := pbmClient.DeleteProfile(ctx, []types.PbmProfileId{*profileID})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+}
+
+//createCategoryNTag create one category and one tag
+func createCategoryNTag(ctx context.Context, catName string, tagName string) (string, string) {
+	tagSpec := tags.Category{Name: catName, Cardinality: "MULTIPLE"}
+	mgr := newTagMgr(ctx, e2eVSphere.Client)
+	catID, err := mgr.CreateCategory(ctx, &tagSpec)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	tagID, err := mgr.CreateTag(ctx, &tags.Tag{Name: tagName, CategoryID: catID})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	return catID, tagID
+}
+
+//deleteCategoryNTag given category and tag
+func deleteCategoryNTag(ctx context.Context, catID string, tagID string) {
+	mgr := newTagMgr(ctx, e2eVSphere.Client)
+
+	tag, err := mgr.GetTag(ctx, tagID)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	err = mgr.DeleteTag(ctx, tag)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+	cat, err := mgr.GetCategory(ctx, catID)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	err = mgr.DeleteCategory(ctx, cat)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+}
+
+//attachTagToDS attach given cat/tag to given datastore(URL)
+func attachTagToDS(ctx context.Context, tagID string, dsURL string) {
+	dcList, err := e2eVSphere.getAllDatacenters(ctx)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	mgr := newTagMgr(ctx, e2eVSphere.Client)
+	var ds *object.Datastore
+	for _, dc := range dcList {
+		ds, err = getDatastoreByURL(ctx, dsURL, dc)
+		if err != nil {
+			if !strings.Contains(err.Error(), "couldn't find Datastore given URL") {
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}
+		} else {
+			break
+		}
+	}
+	err = mgr.AttachTag(ctx, tagID, ds.Reference())
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+}
+
+//detachTagToDS attach given cat/tag to given datastore(URL)
```

Comment:
Please correct the comment saying detaches given cat/tag to given datastore(URL)